### PR TITLE
BAQE-1567 - Change revapi to check against 7.44.0.Final

### PR DESCRIPTION
**Pull request by MarianMacik**
**Thank you for submitting this pull request**

**JIRA**: [BAQE-1567](https://issues.redhat.com/browse/BAQE-1567)

depends on kiegroup/droolsjbpm-build-bootstrap#1485

### CODE REVIEW

File: optaplanner-core/src/build/revapi-config.json [hunk 1]
```diff
@@ -25,68 +25,8 @@
...
-           "package": "org.optaplanner.core.config.score.director",
-           "classSimpleName": "ScoreDirectorFactoryConfig",
-           "methodName": "setIncrementalScoreCalculatorClass",
-           "parameterIndex": "0",
-           "elementKind": "parameter",
-           "justification": "Score calculators moved package."
-         }
-      ]
+      "_comment": "Changes between 7.44.0.Final and the current branch. These changes are desired and thus ignored.",
+      "ignore": []
```
Comment (triceo) [inline optaplanner-core/src/build/revapi-config.json:29]:
```suggestion
      "ignore": [
       {
           "code": "java.method.returnTypeTypeParametersChanged",
           "old": "method java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getEasyScoreCalculatorClass()",
           "new": "method java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getEasyScoreCalculatorClass()",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "getEasyScoreCalculatorClass",
           "elementKind": "method",
           "justification": "Score calculators moved package."
         },
         {
           "code": "java.method.returnTypeTypeParametersChanged",
           "old": "method java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getIncrementalScoreCalculatorClass()",
           "new": "method java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getIncrementalScoreCalculatorClass()",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "getIncrementalScoreCalculatorClass",
           "elementKind": "method",
           "justification": "Score calculators moved package."
         },
         {
           "code": "java.method.parameterTypeParameterChanged",
           "old": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setEasyScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>===)",
           "new": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setEasyScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>===)",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "setEasyScoreCalculatorClass",
           "parameterIndex": "0",
           "elementKind": "parameter",
           "justification": "Score calculators moved package."
         },
         {
           "code": "java.method.parameterTypeParameterChanged",
           "old": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setIncrementalScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>===)",
           "new": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setIncrementalScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>===)",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "setIncrementalScoreCalculatorClass",
           "parameterIndex": "0",
           "elementKind": "parameter",
           "justification": "Score calculators moved package."
         }
    ]
```

Comment (triceo) [reply optaplanner-core/src/build/revapi-config.json:29]: This does actually need to be there.

Comment (MarianMacik) [reply optaplanner-core/src/build/revapi-config.json:29]: Why? It is not needed when I built it locally.

Comment (triceo) [reply optaplanner-core/src/build/revapi-config.json:29]: The change was merged yesterday. Maybe you didn't have the latest code?

Comment (MarianMacik) [reply optaplanner-core/src/build/revapi-config.json:29]: I pulled the latest changes today before noon around 11 am.

Comment (triceo) [reply optaplanner-core/src/build/revapi-config.json:29]: This is incredibly interesting. When I was merging this yesterday, it was necessary. Today, somehow, it is not.
But as you say - if Jenkins is green, everything's OK.

Comment (rsynek) [reply optaplanner-core/src/build/revapi-config.json:29]: @triceo have you tried returning back the <revapi.oldKieVersion>7.39.0.Final</revapi.oldKieVersion> to see if it's still needed as yesterday?

Comment (MarianMacik) [reply optaplanner-core/src/build/revapi-config.json:29]: Yeah, now we check 7.x against 7.44.0.Final, so there may no longer be the violations.

Comment (rsynek) [reply optaplanner-core/src/build/revapi-config.json:29]: That's the interesting part; the changes @triceo pointed to landed on the 7.x branch only this week:
https://github.com/kiegroup/optaplanner/commit/71b35529bec092f5258356188d1afb62c085f681

Comment (MarianMacik) [reply optaplanner-core/src/build/revapi-config.json:29]: Found the culprit.
This time I was much quicker than prod guys :) 7.44.0.Final is not released yet and revapi by default skips check if it cannot download the desired artifact. I will configure it to fail (tested locally), so it doesn't happen anymore. Now we just need to wait till next week, so it is released. Thanks for spotting it.

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by MarianMacik** [comment]
@rsynek can you approve?

**Review by rsynek — APPROVED** [review]
LGTM. Let me comment on the failing PR checks:
- "Build chains": irrelevant failure
- "OptaPlanner CI": does not see other repositories; it has no access to related changes in the kie-parent
- "Linux": irrelevant failure
- "Linux Pull Request": is green and is right as it takes cross-repository PRs into account

**Pull request merged**